controllers/snapshot/snapshot_adapter_test.go

```diff
@@ -700,6 +707,58 @@ var _ = Describe("Snapshot Adapter", Ordered, func() {
 			Expect(buf.String()).Should(ContainSubstring(expectedLogEntry))
 		})
 
+		It("no action when snapshot environmentBinding exists", func() {
+			hasBinding = &applicationapiv1alpha1.SnapshotEnvironmentBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "snapshot-binding-sample",
+					Namespace: "default",
+					Labels: map[string]string{
+						gitops.SnapshotTestScenarioLabel: integrationTestScenario.Name,
+					},
+				},
+				Spec: applicationapiv1alpha1.SnapshotEnvironmentBindingSpec{
+					Application: hasApp.Name,
+					Snapshot:    hasSnapshot.Name,
+					Environment: integrationTestScenario.Spec.Environment.Name,
+					Components:  []applicationapiv1alpha1.BindingComponent{},
+				},
+			}
+			Expect(k8sClient.Create(ctx, hasBinding)).Should(Succeed())
+
+			var buf bytes.Buffer
+			log := helpers.IntegrationLogger{Logger: buflogr.NewWithBuffer(&buf)}
+
+			gitops.MarkSnapshotAsPassed(k8sClient, ctx, hasSnapshot, "test passed")
+			hasSnapshot.Labels[gitops.PipelineAsCodeEventTypeLabel] = gitops.PipelineAsCodePushType
+			Expect(gitops.HaveAppStudioTestsSucceeded(hasSnapshot)).To(BeTrue())
+			adapter = NewAdapter(hasSnapshot, hasApp, hasComp, log, loader.NewMockLoader(), k8sClient, ctx)
+			adapter.context = toolkit.GetMockedContext(ctx, []toolkit.MockData{
+				{
+					ContextKey: loader.ApplicationContextKey,
+					Resource:   hasApp,
+				},
+				{
+					ContextKey: loader.SnapshotContextKey,
+					Resource:   hasSnapshot,
+				},
+				{
+					ContextKey: loader.ApplicationComponentsContextKey,
+					Resource:   []applicationapiv1alpha1.Component{*hasComp},
+				},
+				{
+					ContextKey: loader.SnapshotEnvironmentBindingContextKey,
```

Comment:
We are mocking hasBinding here, do we need to write it down to k8s DB on line 726?

What's the purpose of mocking here if we are writing data into DB?

Reply:
Ok, we need them because snapshot will be written there